contributors/devel/release.md

```diff
@@ -0,0 +1,231 @@
+# Targeting Features, Issues and PRs to Release Milestones
+
+This document is focused on kubernetes developers and contributors
```

Comment:
are we lower-casing kubernetes? It's consistent throughout this doc, but...?

Reply:
https://kubernetes.io/docs/contribute/style/style-guide/#kubernetes-io-word-list says always capitalized.  Fixed throughout doc.  It's not fully consistent elsewhere in the contributor and developer guides.